### PR TITLE
[EAG2-64] Leave game

### PR DESCRIPTION
**Pull request by RayWayy**

### CODE REVIEW

File: src/main/java/com/eleks/academy/whoami/controller/GameController.java [hunk 1]
```diff
@@ -96,4 +96,10 @@ public void answerQuestion(@PathVariable("id") String id,
 
 	}
 
+	@PostMapping("/{id}/leave")
+	public void leaveGame(@PathVariable("id") String id,
+												@RequestHeader(PLAYER) String player) {
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/controller/GameController.java]: fix formatting

File: src/main/java/com/eleks/academy/whoami/service/impl/GameServiceImpl.java [hunk 2]
```diff
@@ -102,4 +103,19 @@ public void answerQuestion(String id, String player, String answer) {
 
 	}
 
+	@Override
+	public void leaveGame(String id, String player) {
+		var game = gameRepository.findById(id).orElseThrow(
+				() -> {
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/service/impl/GameServiceImpl.java]: no need of lambda with {} its one line expression () -> new Exception()

File: src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		int maxPlayer = 4;
+		final Map<String, SynchronousPlayer> players = new HashMap<>(maxPlayer);
+
+		var state = new StateTest(1, maxPlayer, players);
+
+		players.put(player, new PersistentPlayer(player));
+
+		var findPlayer = state.findPlayer(player);
+		var newPlayer = new PersistentPlayer(player);
+		Optional<SynchronousPlayer> expectedPlayer = Optional.of(newPlayer);
```
Comment (VolodymyrKr-Eleks) [inline src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java]: easier to get from optional than to wrap something into optional

**Review by VolodymyrKr-Eleks — CHANGES_REQUESTED** [review]

File: src/main/java/com/eleks/academy/whoami/controller/GameController.java (hunk 1 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/controller/GameController.java]: Resolved

File: src/main/java/com/eleks/academy/whoami/service/impl/GameServiceImpl.java (hunk 2 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/service/impl/GameServiceImpl.java]: Resolved

File: src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java (hunk 3 above)
Comment (RayWayy) [reply src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java]: Resolved

File: src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java [hunk 4]
```diff
@@ -23,9 +22,9 @@ void findPlayerTest() {
 
 		players.put(player, new PersistentPlayer(player));
 
-		var findPlayer = state.findPlayer(player);
-		var newPlayer = new PersistentPlayer(player);
-		Optional<SynchronousPlayer> expectedPlayer = Optional.of(newPlayer);
+		var findPlayerOp = state.findPlayer(player);
```
Comment (VolodymyrKr-Eleks) [inline src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java]: @Flozik look: `assertEquals(new PersistentPlayer(player), state.findPlayer(player).get());` and I even do not create variables, variables are only needed when you going to use them somewhere if you can do without them do it

**Review by VolodymyrKr-Eleks — CHANGES_REQUESTED** [review]

**Pull request merged**